htlcswitch/link.go

```diff
@@ -3421,6 +3444,26 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 				BlindingPoint: fwdInfo.NextBlinding,
 			}
 
+			err = fn.MapOptionZ(
+				pd.CustomRecords, func(b tlv.Blob) error {
+					r, err := lnwire.ParseCustomRecords(b)
+					if err != nil {
+						return err
+					}
+
+					addMsg.CustomRecords = r
```

Comment:
Do we want to be overloading the `CustomRecords` field here?
* `pd.CustomRecords` are the records we received on the _incoming_ HTLC's `update_add_htlc`
* `addMsg.CustomRecords` are the records that we're going to pack for the _outgoing_ HTLC's `update_add_htlc`

This has the impact of always relaying *all* of the custom TLVs that we get on the incoming `update_add_htlc` to the outgoing `update_add_htlc` -  confirmed this with a [quick itest hack](https://github.com/lightningnetwork/lnd/files/15380880/copy_tlvs.patch) - and I don't see anything in the original issue saying that we want to do this?

Reply:
nice catch
Agree, we don't want this side effect of naively relaying everything

I believe we actually want to block any type of relaying. These records should be propagated all the way until the interception point, where the user may decide if any outgoing htlc wire records will be defined

Reply:
Assumed not, can prob easily fix by just adding a separate field to the intercepted packet.

Reply:
I think this was actually introduced in a previous PR () that added the interceptor logic. I'll address this in a later PR, added it to the tracking issue for now: https://github.com/lightningnetwork/lnd/issues/8769
We really are blocked by this PR (which is just one piece in a long series of PRs), so I'll want to merge this into the staging branch asap and then address any remaining issues later.

Reply:
The previous PR is correctly using this field to `ResumeModified` with `CustomRecords` that are provided by the interceptor. 

Here, we're overloading it with the incoming `update_add_htlc`'s `CustomRecords` (which introduces the issue). The fix is pretty simple - just add a field to `htlcPacket` and surface it separately on the interceptor?

Reply:
[Patch](https://github.com/lightningnetwork/lnd/files/15403225/incomingrecords.patch) to fix is trivial, [messy test](https://github.com/lightningnetwork/lnd/files/15403227/recordstest.patch) to confirm that it solves the problem.

Reply:
> Thanks a lot for the patch

❤️ it needs `gofmt` :')